### PR TITLE
Cleanup and improve `--check-cfg` implementation

### PR DESCRIPTION
**Pull request by Urgau**
This PR removes some indentation in the code, as well as preventing some bugs/misusages and fix a nit in the doc.

r? @petrochenkov (maybe)

### CODE REVIEW

File: compiler/rustc_interface/src/interface.rs [hunk 1]
```diff
@@ -139,226 +139,209 @@ pub fn parse_check_cfg(handler: &EarlyErrorHandler, specs: Vec<String>) -> Check
...
-                },
-                Err(errs) => {
-                    drop(errs);
-                    expected_error();
                 }
+            } else {
+                expected_error();
             }
+
+            drop(parser);
```
Comment (petrochenkov) [inline compiler/rustc_interface/src/interface.rs:361]: `parser` is dropped here automatically, no?

**Comment by petrochenkov** [comment]
r=me with https://github.com/rust-lang/rust/pull/117025#discussion_r1373527915 addressed.
@rustbot author

File: compiler/rustc_interface/src/interface.rs (hunk 1 above)
Comment (Urgau) [reply compiler/rustc_interface/src/interface.rs]: :confused: apparently yes, but when I refactored it, it was definitely mad at me if I didn't include the `drop(parser)`.

Anyway, removed.

**Comment by Urgau** [comment]
[#117025 (comment)](https://github.com/rust-lang/rust/pull/117025#discussion_r1373527915) has been addressed

@rustbot ready
*(since I don't have bors rights)*

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang-ci/rust/actions/runs/6663550445/job/18109620159#step:25:29546
failed in https://github.com/rust-lang/rust/pull/117264

**Comment by Urgau** [comment]
The `libc` crate is using custom check-cfg, but it's [using the old-deprecated syntax](https://github.com/rust-lang-ci/rust/actions/runs/6663550445/job/18109620159#step:25:29548) preventing the anti-mixing check to be enabled.

We could disable those extra libc check-cfg, by [unsetting the `LIBC_CHECK_CFG` env](https://github.com/rust-lang/rust/blob/707d8c3f1bba10d5aaa9c8719c410711edf2e227/src/bootstrap/src/core/builder.rs#L1883), but this might break downstream forks using custom patched libc. So until the libc crate is updated to support the new syntax, I've dropped the anti-mixing check (for now).

@rustbot ready

**Comment by petrochenkov** [comment]
>I've dropped the anti-mixing check (for now).

I don't think that check is very important, the old syntax will be removed anyway and mixing will go away with it.

@bors r+

**Pull request merged**